contracts/contracts/coordination/TACoChildApplication.sol

```diff
@@ -123,6 +130,56 @@ contract TACoChildApplication is ITACoRootToChild, ITACoChildApplication, Initia
         emit OperatorConfirmed(stakingProvider, _operator);
         rootApplication.confirmOperatorAddress(_operator);
     }
+
+    /**
+     * @notice Return the length of the array of staking providers
+     */
+    function getStakingProvidersLength() external view returns (uint256) {
+        return stakingProviders.length;
+    }
+
+    /**
+     * @notice Get the value of authorized tokens for active providers as well as providers and their authorized tokens
+     * @param _startIndex Start index for looking in providers array
+     * @param _maxStakingProviders Max providers for looking, if set 0 then all will be used
+     * @return allAuthorizedTokens Sum of authorized tokens for active providers
+     * @return activeStakingProviders Array of providers and their authorized tokens.
+     * Providers addresses stored as uint256
+     * @dev Note that activeStakingProviders[0] is an array of uint256, but you want addresses.
+     * Careful when used directly!
+     */
+    function getActiveStakingProviders(
+        uint256 _startIndex,
+        uint256 _maxStakingProviders
+    )
+        external
+        view
+        returns (uint256 allAuthorizedTokens, uint256[2][] memory activeStakingProviders)
+    {
+        uint256 endIndex = stakingProviders.length;
+        require(_startIndex < endIndex, "Wrong start index");
+        if (_maxStakingProviders != 0 && _startIndex + _maxStakingProviders < endIndex) {
+            endIndex = _startIndex + _maxStakingProviders;
+        }
+        activeStakingProviders = new uint256[2][](endIndex - _startIndex);
+        allAuthorizedTokens = 0;
+
+        uint256 resultIndex = 0;
+        for (uint256 i = _startIndex; i < endIndex; i++) {
+            address stakingProvider = stakingProviders[i];
+            StakingProviderInfo storage info = stakingProviderInfo[stakingProvider];
+            uint256 eligibleAmount = info.authorized;
+            if (eligibleAmount < minimumAuthorization || !info.operatorConfirmed) {
+                continue;
+            }
+            activeStakingProviders[resultIndex][0] = uint256(uint160(stakingProvider));
```

Comment:
Just out of curiosity: why returning `uint256` instead of `address`?

Reply:
Because it's an array with 2 columns, one for address and the other for amount, so `uint256` seems like the more "common" type. 

@vzotova knowing that the address plus the amount (uint96) fit exactly in 1 word, how about returning a single uint256 value? This has the advantage that it would reduce the likelihood of using pagination by 50%, at the expense of requiring a very minor processing on the client side.

Reply:
✔️